sdk/src/wallet/operations/balance.rs

```diff
@@ -271,6 +285,7 @@ where
                         output_data.output_id.transaction_id().slot_index(),
                         slot_index,
                     )?;
+                    mana_rewards.remove(&output_data.output_id);
```

Comment:
Shouldn't the rewards stay if it's an account or the first created delegation output? Because just transition these should not remove the rewards?

Reply:
Can you explain more?

Reply:
If you have a staking account and do a tx where you just transition the account, this doesn't affect the rewards at all? Or for delegation, if you transition into delayed delegation, this also changes nothing for the rewards?

Reply:
but these are just locked outputs?

Reply:
Locked output means in this case that they're used as input in pending transactions

Reply:
right so we can't really know if those rewards are claimed right

Reply:
Also even if the tx would claim them, as long as it's not confirmed, I see no reason to not still show the rewards

Reply:
isn't that true of all locked outputs though?

Reply:
Yes, actually would never remove any the rewards

Reply:
If you say so 🤷‍♀️